### PR TITLE
ZKUI-129: Prefix bucket name in create new rule form 

### PR DESCRIPTION
**Pull request by ChengYanJin**
- Prefix bucket name in create new rule form 
- Redirect back to bucket list if create a new rule from bucket->workflow

### CODE REVIEW

File: src/react/workflow/ExpirationForm.tsx [hunk 1]
```diff
@@ -127,176 +127,141 @@ export function ExpirationForm({ bucketList, locations, prefix = '' }: Props) {
...
+                      }
+                      const defaultBucketOption = options.find(
+                        (op) => op.value === bucketName,
+                      );
+                      return (
+                        <Select
+                          id="sourceBucket"
+                          value={sourceBucket}
+                          onChange={onChange}
+                          defaultValue={defaultBucketOption}
```
Comment (JBWatenbergScality) [inline src/react/workflow/ExpirationForm.tsx]: shouldn't it just be the bucket name here as a default value ? I'm a bit surprised that it works with the "option" object

File: src/react/workflow/ReplicationForm.tsx [hunk 2]
```diff
@@ -185,13 +194,25 @@ function ReplicationComponent({
...
+                      (op) => op.value === bucketName,
+                    );
                     return (
-                      <Select id="sourceBucket" value={sourceBucket} onChange={onChange} isDisabled={isEditing}>
+                      <Select
+                        id="sourceBucket"
+                        value={sourceBucket}
+                        onChange={onChange}
+                        isDisabled={isEditing}
+                        defaultValue={defaultBucketOption}
```
Comment (JBWatenbergScality) [inline src/react/workflow/ReplicationForm.tsx]: Same as for expiration

File: src/react/workflow/utils.tsx [hunk 3]
```diff
@@ -70,29 +71,29 @@ export const renderDestination = (locations: Locations) => {
...
+    workflowId: '',
+    currentVersionTriggerDelayDate: '',
+    currentVersionTriggerDelayDays: null,
+    expireDeleteMarkersTrigger: null,
+    incompleteMultipartUploadTriggerDelayDays: null,
+    previousVersionTriggerDelayDays: null,
+  };
 }
-export function newReplicationForm(): ReplicationForm {
+export function newReplicationForm(bucketName: string): ReplicationForm {
```
Comment (JBWatenbergScality) [inline src/react/workflow/utils.tsx]:
```suggestion
export function newReplicationForm(bucketName?: string): ReplicationForm {
```

File: src/react/workflow/utils.tsx [hunk 4]
```diff
@@ -70,29 +71,29 @@ export const renderDestination = (locations: Locations) => {
     );
   };
 };
-export function newExpiration(): Expiration {
+export function newExpiration(bucketName: string): Expiration {
```
Comment (JBWatenbergScality) [inline src/react/workflow/utils.tsx]:
```suggestion
export function newExpiration(bucketName?: string): Expiration {
```

File: src/react/workflow/ExpirationForm.tsx (hunk 1 above)
Comment (ChengYanJin) [reply src/react/workflow/ExpirationForm.tsx]: Indeed I figured I don't need to add any defaultValue here ..

**Review by JBWatenbergScality — DISMISSED** [review]

File: src/react/workflow/utils.tsx [hunk 5]
```diff
@@ -88,12 +88,12 @@ export function newExpiration(bucketName: string): Expiration {
...
   };
 }
-export function newReplicationForm(bucketName: string): ReplicationForm {
+export function newReplicationForm(bucketName?: string): ReplicationForm {
   return {
     streamVersion: 1,
     streamId: '',
     enabled: true,
-    sourceBucket: bucketName || '',
+    sourceBucket: bucketName | '',
```
Comment (JBWatenbergScality) [inline src/react/workflow/utils.tsx]: ||

**Review by JBWatenbergScality — APPROVED** [review]

**Pull request merged**